pennylane/workflow/get_gradient_fn.py

```diff
@@ -61,8 +61,16 @@ def _get_gradient_fn(
         )
 
     if diff_method == "best":
-        qn = qml.QNode(lambda: None, device, diff_method=None)
-        return qml.workflow.get_best_diff_method(qn)()
+        config = _make_execution_config(None, "best")
+
+        if device.supports_derivatives(config, circuit=tape):
+            new_config = device.preprocess(config)[1]
+            return new_config.gradient_method
+
+        if tape and any(isinstance(o, qml.operation.CV) for o in tape):
+            return qml.gradients.param_shift_cv
+
+        return qml.gradients.param_shift
```

Comment:
Moved this logic from `qml.workflow.get_best_diff_method`. No longer need to create a dummy `QNode`; which prevented me using this helper function in `qnode.py`